app/uk/gov/hmrc/merchandiseinbaggage/service/EmailService.scala

```diff
@@ -139,38 +149,60 @@ class EmailService @Inject()(emailConnector: EmailConnector, declarationReposito
       "eori"                      -> eori.value
     )
 
-    val calculationParams = {
-      maybeTotalCalculationResult match {
-        case Some(total) =>
-          Map(
-            "customsDuty" -> total.totalDutyDue.formattedInPounds,
-            "vat"         -> total.totalVatDue.formattedInPounds,
-            "total"       -> total.totalTaxDue.formattedInPounds
-          )
-
-        case None => Map.empty
-      }
-    }
-
     val allParams =
       if (declarationType == DeclarationType.Import)
-        goodsParams ++ commonParams ++ calculationParams
+        goodsParams ++ commonParams ++ paymentParams(declaration)
       else
         goodsParams ++ commonParams
 
     DeclarationEmailInfo(
       Seq(emailTo),
-      templateId(lang, declarationType),
+      templateId(lang, declarationType, amendments),
       allParams
     )
   }
 
-  private def templateId(lang: String, declarationType: DeclarationType): String = {
-    val importTemplate = if (lang == "en") "mods_import_declaration" else "mods_import_declaration_cy"
-    val exportTemplate = if (lang == "en") "mods_export_declaration" else "mods_export_declaration_cy"
-    if (declarationType == DeclarationType.Import)
-      importTemplate
-    else
-      exportTemplate
+  private def templateId(lang: String, declarationType: DeclarationType, amendments: Seq[Amendment]): String = {
+    val isAmendJourney = amendments.nonEmpty
+
+    val templateId = (declarationType, isAmendJourney) match {
```

Comment:
is it hard to pass journeyType so you can pattern match on Amend & New ? Instead of booleans

Reply:
i didn't want to do the pattern match inside the templates as we may end up with too many such checks... 
instead i have created separate amend templates...
https://github.com/hmrc/hmrc-email-renderer/pull/714

Reply:
??? you're pattern matching or am i missing something.....??? I was saying to use ADT Amend&New instead of Boolean
  case (Import, false) => "mods_import_declaration"
      case (Import, true)  => "mods_amend_import_declaration"
      case (Export, false) => "mods_export_declaration"
      case (Export, true)  => "mods_amend_export_declaration"
    }

Reply:
ohh you mean here ... yeah, i thought we only use this boolean check once in this method.. but happy to add an ADT to make it cleaner...